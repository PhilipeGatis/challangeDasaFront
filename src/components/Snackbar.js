import React, { useEffect, useState } from 'react';
import { useLocation, useHistory } from 'react-router-dom';
import { Alert } from 'antd';
import { parse, stringify } from 'qs';
import styled from 'styled-components';

const StyledAlert = styled(Alert)`
  position: absolute;
  bottom: 5px;
  width: 100%;
`;

const Snackbar = () => {
  const [showAlert, toggleAlert] = useState(false);
  const location = useLocation();
  const history = useHistory();
  const { search } = location;
  const { error, ...restQs } = parse(search, { ignoreQueryPrefix: true });

  useEffect(() => {
    if (error) {
      toggleAlert(true);
    }
  }, [error]);

  useEffect(() => {
    if (!showAlert) return;
    const timer = setTimeout(() => {
      toggleAlert(false);
      clearTimeout(timer);
      history.replace({
        pathname: location.pathname,
        search: stringify(restQs),
      });
    }, 3000);

    return () => clearTimeout(timer);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [showAlert]);

  return showAlert ? <StyledAlert type="error" message={error} banner /> : null;
};

export default Snackbar;
