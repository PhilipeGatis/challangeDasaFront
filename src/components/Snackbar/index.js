import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import { Alert } from 'antd';
import { parse } from 'qs';
import styled from 'styled-components';

const StyledAlert = styled(Alert)`
  position: absolute;
  bottom: 5px;
  width: 100%;
`;

const Snackbar = () => {
  const [showAlert, toggleAlert] = useState(false);
  const location = useLocation();
  const { search } = location;
  const { error } = parse(search, { ignoreQueryPrefix: true });

  useEffect(() => {
    if (error) {
      toggleAlert(true);
    }
  }, [error]);

  useEffect(() => {
    if (!showAlert) return;
    const timer = setTimeout(() => {
      toggleAlert(false);
      clearTimeout(timer);
    }, 3000);

    return () => clearTimeout(timer);
  }, [showAlert]);
  return showAlert ? <StyledAlert type="error" message={error} banner /> : null;
};

export default Snackbar;
