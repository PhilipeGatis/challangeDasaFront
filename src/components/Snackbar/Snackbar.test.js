import React from 'react';
import Snackbar from './Snackbar';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<Snackbar />', () => {
  const RenderWithRouter = ({ children }) => (
    <MemoryRouter initialEntries={['/?error=errorMessage']}>
      <Route path="/">{children}</Route>
    </MemoryRouter>
  );

  it('should render', async () => {
    const wrapper = render(
      <RenderWithRouter>
        <Snackbar />
      </RenderWithRouter>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
