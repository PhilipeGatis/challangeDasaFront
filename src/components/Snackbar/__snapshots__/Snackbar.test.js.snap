// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should render 1`] = `null`;
